src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java

```diff
@@ -121,8 +143,85 @@ public static void setDashboards(String user, Set<String> projects, Set<String>
 				if ( common.size() > 0 )
 					setDefaultIndex(user, common.get(0), esClient, kibanaIndex, kibanaVersion);
 			}
+		}
+		
+		if ( changed )
+			refreshKibanaUser(user, kibanaIndex, esClient);
+	}
+	
+	private static void refreshKibanaUser(String username, String kibanaIndex, Client esClient) {
+		
+		String userIndex = getKibanaIndex(username, kibanaIndex);
+		RefreshRequest request = new RefreshRequest().indices(userIndex);
+		RefreshResponse response = esClient.admin().indices().refresh(request).actionGet();
+		
+		logger.debug("Refreshed '{}' successfully on {} of {} shards", userIndex, response.getSuccessfulShards(), response.getTotalShards());
+	}
+	
+	private static void initialKibanaSearchMapping(String username, String kibanaIndex, Client esClient) {
+		
+		String userIndex = getKibanaIndex(username, kibanaIndex);
+		try {
+			GetIndexResponse getResponse = esClient.admin().indices().getIndex(new GetIndexRequest().indices(userIndex)).actionGet();
+			
+			logger.debug("getIndexResponse mapping '{}'", getResponse.getMappings().get(userIndex).keys() ); 
+			
+			if ( !getResponse.getMappings().get(userIndex).containsKey("search") ) {
+			
+				PutMappingRequest putRequest = new PutMappingRequest().indices(new String[]{userIndex}).type("source").source(
+				"{\"properties\":{\"title\":{\"type\":\"string\"},\"description\":{\"type\":\"string\"},\"hits\":{\"type\":\"integer\"},\"columns\":{\"type\":\"string\"},\"sort\":{\"type\":\"string\"},\"version\":{\"type\":\"integer\"},\"kibanaSavedObjectMeta\":{\"properties\":{\"searchSourceJSON\":{\"type\":\"string\"}}}}}"		
```

Comment:
I wonder if we should move this to a file for easier updating and editing.  We could statically load it once and then it would mitigate having to do all the escaping